config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-presubmits.yaml

```diff
@@ -209,3 +223,67 @@ presubmits:
       testgrid-tab-name: pr-e2e
       testgrid-alert-email: k8s-testing-clusterapi-vsphere+alerts@groups.vmware.com
       description: Runs e2e tests
+
+postsubmits:
+  kubernetes-sigs/cluster-api-provider-vsphere:
+
+  # Deploys images and binaries after all merges to master
+  - name: post-cluster-api-provider-vsphere-deploy
+    labels:
+      preset-dind-enabled: "true"
+      preset-cluster-api-provider-vsphere-gcs-creds: "true"
+    branches:
+    - ^master$
+    always_run: false
+    run_if_changed: '^(cmd|config|pkg|vendor)/|Dockerfile'
+    decorate: true
+    path_alias: sigs.k8s.io/cluster-api-provider-vsphere
+    max_concurrency: 1
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20190719-7cdf877-master
+        command:
+        - runner.sh
+        args:
+        - ./hack/release.sh
+        - -l
+        - -p
+        - -k
+        - /root/.capv/keyfile.json
+        # we need privileged mode in order to do docker in docker
+        securityContext:
+          privileged: true
+    annotations:
+      testgrid-dashboards: vmware-cluster-api-provider-vsphere, sig-cluster-lifecycle-cluster-api-provider-vsphere
+      testgrid-tab-name: post-deploy
+      testgrid-alert-email: k8s-testing-clusterapi-vsphere+alerts@groups.vmware.com
+      description: Pushes new images and binaries from master
+
+  # Deploys images and binaries for tagged releases
+  - name: post-cluster-api-provider-vsphere-release
+    labels:
+      preset-dind-enabled: "true"
+      preset-cluster-api-provider-vsphere-gcs-creds: "true"
+    branches:
+    - ^v(\d)+\.(\d)+\.(\d)+(-(alpha|beta|rc)\.(\d)+)?$
```

Comment:
Hi @codenrhoden,

You probably know this already, but does Prow use an annotated tag as the branch name? Or is that something that must be checked in the release script? In other words, will this work as expected?

**Update**
I just searched the Prow repo and found [this](https://github.com/kubernetes/test-infra/blob/aba9431414f46da9080d1d1b92a8c7da51b772f7/config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml#L48-L49):

```yaml
  - name: post-sig-storage-local-static-provisioner-build-release
    branches:
    - ^v\d+\.\d+\.\d+$ # stable
    - ^v\d+\.\d+\.\d+(-[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$ # pre-release
```

So it *does* appear a pattern that matches tags should work. I also [confirmed](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner/branches) that none of the project's branches match that pattern, so it **must** be for use with tags.

Reply:
Yep, that's the whole point around this. This is how we get Prow to react to annotated tags.